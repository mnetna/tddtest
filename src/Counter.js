import React, {useState} from 'react';

const Counter = () => {
  const [value, setValue] = useState(0);

  const increase = () => {
    setValue(value + 1);
  }

  const decrease = () => {
    setValue(value - 1);
  }

  return (
    <div data-test="component-app"> 
      <p>
        <b>{value}</b>
      </p>
      <button id="increase" onClick={increase}>+1</button>
      <button id="decrease" onClick={decrease}>-1</button>
    </div>
  );
};

export default Counter;