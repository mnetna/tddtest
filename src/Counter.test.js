import React from 'react';
import Enzyme, { mount } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Counter from './Counter';

Enzyme.configure({ adapter: new EnzymeAdapter() });

const wrapper = mount(<Counter />);
const appComponent = wrapper.find('[data-test="component-app"]');

test("에러 없이 렌더링이 잘된다.", () => {
  expect(appComponent).toMatchSnapshot();
});

test("플러스 버튼 렌더링이 잘된다. ", () => {
  expect(appComponent.find('#increase')).toHaveLength(1);
});

test("카운터 숫자가 렌더링이 잘된다.", () => {
  expect(appComponent.find('b')).toHaveLength(1);
});

test("카운터의 초기값은 0이다.", () => {
  expect(appComponent.find('b').text()).toEqual('0');
});

test("클릭 버튼을 누르면 카운터 숫자가 증가된다.", () => {
  let plusButton = appComponent.find('#increase');
  plusButton.simulate('click');
  let number = appComponent.find('b');
  expect(number.text()).toBe('1');
});