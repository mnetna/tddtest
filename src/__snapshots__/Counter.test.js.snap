// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`에러 없이 렌더링이 잘된다. 1`] = `ReactWrapper {}`;
